packages/plugin-ext/src/api/rpc-protocol.ts

```diff
@@ -26,6 +26,7 @@ import { Event } from '@theia/core/lib/common/event';
 import { Deferred } from '@theia/core/lib/common/promise-util';
 import VSCodeURI from 'vscode-uri';
 import URI from '@theia/core/lib/common/uri';
+import { CancellationToken, CancellationTokenSource } from 'vscode-jsonrpc/lib/cancellation';
```

Comment:
it's kind of minor but shouldn't we import from vscode-languageserver-protocol instead of vscode-jsonrpc  (I know it's defined in vscode-jsonrpc but it's odd here to depends from jsonrpc instead of the protocol vscode-languageserver-protocol

https://github.com/Microsoft/vscode-languageserver-node/blob/dbe64946d7e41a50996a7c2929ac4aa4911b505d/protocol/src/main.ts#L21

Reply:
also what's about https://github.com/theia-ide/theia/blob/cd1c462e7585b6f1b3956fc5db2e6570f6789f45/packages/core/src/common/cancellation.ts ?

Reply:
I want to remove dependencies to theia extensions from plugin host process as much as possible. Will replace with import from protocol

Reply:
> remove dependencies to theia extensions

it includes `@theia/core` as well ?

Reply:
An idea was to allow spawning plugin host process over ssh in another container. `@theia/core` should not be necessary then to install at build time and load at runtime. It's another issue though I can import from `@theia/core` as well for now if you like it more.

Reply:
I'm fine to limit dependencies (it's good). But I'm still wondering which package need to be 'cut off'
Like reducing core dependencies, or reducing external dependencies vscode-* dependencies ?

I think the goal is to have the smallest bootstrap but I don't know which path is better

> I can import from @theia/core as well for now if you like it more.

I'm fine with any dependencies

Reply:
`@theia/core` has quite a lot of unrelated code and pulls difference dependencies like phosphor. I would prefer not installing it.

In Che as far as I understood you deploy Theia in each side container to serve a VS Code extension. I guess it would be interesting as well just deploy code only for host process and wrap it in node server without `@theia/core`.

Reply:
@akosyakov yes I share the same concern, I just wanted to be sure. (for me there is phosphor but also electron right now) and yes many unwanted stuff. I don't know when to start the split of plugin-ext module as it's highly changed right now

Reply:
We've removed all electron deps btw:
https://github.com/theia-ide/theia/pull/4873
https://github.com/theia-ide/theia/pull/4882
https://github.com/theia-ide/theia/pull/4885
https://github.com/theia-ide/theia-apps/pull/165

Reply:
ok thanks, I'm still reading mails/changes after one week of PTO :)